ui/src/app/store/general/utils/certificates.ts

```diff
@@ -0,0 +1,16 @@
+import type { CertificateMetadata } from "app/store/general/types";
+
+export const splitCertificateName = (
+  certificateName: CertificateMetadata["CN"] | null
+): { name: string; host: string } | null => {
+  if (!certificateName) {
+    return null;
+  }
+  const split = certificateName.split("@");
+  if (split.length < 2) {
+    return { name: certificateName, host: "" };
+  }
+  const host = split[split.length - 1];
+  const name = split.slice(0, split.length - 1).join("@");
```

Comment:
It might be good to leave a comment about what this is extracting/what the format is that the data is in.

Reply:
Done!